### PR TITLE
draft for listing a duration of analysis

### LINKED ISSUE #147: We should list duration of analysis
In the analysis list please show the duration of each analysis. It will be useful for debugging slow processing workflows and help to manage user expectations as to how long they should wait for their jobs.

screen shot 2017-06-24 at 10 00 50 pm

### PR DESCRIPTION
**Pull request by boney-bun**
this commit is unfinished work

push to be reviewed

fix #147

### CODE REVIEW

**Comment by boney-bun** [comment]
@lucernae @ismailsunni i need your opinion on how to calculate the duration.
the proposed start time is on the `process_impact_result` function in `analysis.py`
while, the proposed end time is still on the same function as above, but after the `task_state = 'SUCCESS'`

File: models.py [hunk 1]
```diff
@@ -181,6 +180,16 @@ class Meta:
         upload_to='analysis/report/'
     )
 
+    start_time = models.CharField(
```
Comment (ismailsunni) [inline models.py]: Why don't you use `DateTimeField`?

**Review by ismailsunni — CHANGES_REQUESTED** [review]
IMHO, it's better to store it as DateTimeField, then do the formating in the jinja (e.g. https://stackoverflow.com/questions/4830535/python-how-do-i-format-a-date-in-jinja2)

By doing this, we can also calculate the elapsed time, or filter based on the date / month, etc.


UPDATE:
@ismailsunni the charField has been changed to DateTimeField.

**Comment by boney-bun** [comment]
the code is fully working now. below is a sample snapshot:
One thing that puzzle me: at which points should i calculate the duration?
i've got 5 hours duration if the calculation starts from `Date Created` until `end_time`.
interestingly, i would get 0 minutes if the calculation starts from `start_time` to `end_time`.

- `Date Created` is obtained when impact layers is created for the first time.
- `start_time` is marked when the analysis is saved.
- `end_time` is marked when the `task_state` is `SUCCESS`

i'm thinking to record the `end_time` in function `run_analysis` on `inasafe.headless.tasks.inasafe_wrapper.py`. 
what do you think @ismailsunni @lucernae ?

screen shot 2017-08-08 at 1 03 21 am

**Comment by gubuntu** [comment]
I don't think we need to show start and end time, just duration. And the duration above doesn't match the start and end times shown. 

Otherwise LGTM

**Comment by boney-bun** [comment]
The `duration` field is added to the successful analyses. 
otherwise,`start time` field is used (at this point, there has no infomation about `end time` yet so we can't have a duration):

screen shot 2017-08-14 at 4 32 08 pm

Later, when the analysis succeed, the `start time` information would change to `duration` field.
screen shot 2017-08-14 at 4 33 51 pm

**Note:**
Even the seconds is taken into account on the time calculation.
0 minutes on the above image because the elapsed time is less than a minute.
the image below is provided as an illustration that it was not actually *zero minute*:
screen shot 2017-08-14 at 4 33 31 pm

**Comment by boney-bun** [comment]
i'll close this ticket and will merge to staging soon. 
feel free to reopen again.

**Pull request merged**